packages/core/src/physics/PhysicsScene.ts

```diff
@@ -374,10 +378,161 @@ export class PhysicsScene {
+  }
+
+  /**
+   * Casts a sphere through the Scene and returns true if there is any hit.
+   * @param center - The center of the sphere
+   * @param radius - The radius of the sphere
+   * @param direction - The direction to sweep along
+   * @param distance - The max distance to sweep, default is Number.MAX_VALUE
+   * @param layerMask - Layer mask that is used to selectively ignore Colliders when sweeping, default is Layer.Everything
+   * @param outHitResult - If true is returned, outHitResult will contain more detailed collision information
+   * @returns Returns True if the sphere intersects with any collider, otherwise false
+   */
+  sphereCast(
+    center: Vector3,
+    radius: number,
+    direction: Vector3,
+    distance?: number,
+    layerMask?: Layer,
+    outHitResult?: HitResult
+  ): boolean {
+    const sphereGeometry = Engine._nativePhysics.createSphereGeometry(radius);
+    const pose = PhysicsScene._tempPose;
+    pose.translation.copyFrom(center);
+    const result = this._sweep(sphereGeometry, pose, direction, distance, layerMask, outHitResult);
+    sphereGeometry.release();
+    return result;
+  }
+
+  /**
+   * Casts a capsule through the Scene and returns true if there is any hit.
+   * @param center - The center of the capsule
+   * @param radius - The radius of the capsule
+   * @param height - The height of the capsule
+   * @param direction - The direction to sweep along
+   * @param orientation - The rotation of the capsule
+   * @param distance - The max distance to sweep, default is Number.MAX_VALUE
+   * @param layerMask - Layer mask that is used to selectively ignore Colliders when sweeping, default is Layer.Everything
+   * @returns Returns True if the capsule intersects with any collider, otherwise false
+   */
+  capsuleCast(
+    center: Vector3,
+    radius: number,
+    height: number,
+    orientation: Quaternion,
+    direction: Vector3,
+    distance?: number,
+    layerMask?: Layer,
+    outHitResult?: HitResult
+  ): boolean {
+    const capsuleGeometry = Engine._nativePhysics.createCapsuleGeometry(radius, height);
+    const pose = PhysicsScene._tempPose;
+    pose.translation.copyFrom(center);
+    pose.rotation.copyFrom(orientation);
+    const result = this._sweep(capsuleGeometry, pose, direction, distance, layerMask, outHitResult);
+    capsuleGeometry.release();
+    return result;
+  }
+
+  /**
+   * Check if a box overlaps with any collider in the scene.
+   * @param center - The center of the box
+   * @param orientation - The rotation of the box
+   * @param halfExtents - Half the size of the box in each dimension
+   * @param layerMask - Layer mask that is used to selectively ignore Colliders when checking, default is Layer.Everything
+   * @returns Returns True if the box overlaps with any collider, otherwise false
+   */
+  overlapBox(
+    center: Vector3,
+    orientation: Quaternion,
+    halfExtents: Vector3,
+    layerMask?: Layer,
+    outOverlapHitResult?: OverlapHitResult
```

Comment:
The core reason you built this structure is to return the desired Entity, but the reusability of Shape increases the complexity of this function.